src/util/ieee_float.cpp

```diff
@@ -1099,9 +1099,9 @@ void ieee_floatt::from_float(const float f)
 {
   spec.f=23;
   spec.e=8;
-  assert(spec.width()==32);
+  PRECONDITION(spec.width()==32);
 
-  assert(sizeof(float)==sizeof(unsigned int));
+  PRECONDITION(sizeof(float)==sizeof(unsigned int));
 
   union
   {
```

Comment:
This one is subtle. Should be safe. If intentional. 
As these are built-in types, their memory is allocated in one block. (As for PODs). The c++ spec is not entirely clear on this one. 
At the end of the day one of the key purposes of a union is saving on memory - so this is a scenario that was considered as part of c design. Both types will start at the same memory location. 
Not much inclination to check how intentional/w consequences this is - this would be known as 'type punning'  in C - you can essentially 'slice' a type.